File: check_data.py
import os
import json
from datetime import datetime
from db import get_client

db = get_client()

# 6/2 데이터 확인
res = db.table("post_daily_stats").select("*").eq("measured_at", "2026-06-02").execute()
stats = res.data or []

print(f"\n📊 6/2 일일 통계: {len(stats)}건")
print("=" * 80)

total_plays = sum(s.get("play_count", 0) for s in stats if s.get("play_count"))
print(f"총 조회수: {total_plays:,}")
print()

# 상위 10개 게시물
sorted_stats = sorted(stats, key=lambda s: s.get("play_count", 0), reverse=True)[:10]
for i, stat in enumerate(sorted_stats, 1):
    # 게시물 정보 조회
    post_res = db.table("sponsored_posts").select("*").eq("id", stat["post_id"]).execute()
    post = post_res.data[0] if post_res.data else {}
    
    print(f"{i}. {post.get('account_name', '?')} - {stat.get('play_count', 0):,} 조회")
    print(f"   URL: {post.get('url', '?')}")
    print()

# 다른 날짜 비교
print("\n📅 전체 기간 일별 조회수 합계:")
print("=" * 80)
for date in ["2026-05-31", "2026-06-01", "2026-06-02", "2026-06-03", "2026-06-04", "2026-06-05"]:
    res = db.table("post_daily_stats").select("play_count").eq("measured_at", date).execute()
    total = sum(s.get("play_count", 0) for s in (res.data or []) if s.get("play_count"))
    count = len([s for s in (res.data or []) if s.get("play_count")])
    print(f"{date}: {total:,} (건: {count})")

File: check_data.js
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config({ path: 'web/.env.local' });

const supabase = createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL,
  process.env.SUPABASE_SERVICE_ROLE_KEY
);

(async () => {
  console.log('6/1-6/2 데이터 조회 중...\n');

  // 6/1-6/2의 모든 일일 통계 조회
  const { data, error } = await supabase
    .from('post_daily_stats')
    .select('measured_at, play_count, sponsored_posts(id, account_name, posted_at)')
    .gte('measured_at', '2026-06-01')
    .lte('measured_at', '2026-06-02')
    .order('measured_at', { ascending: false })
    .limit(1000);

  if (error) {
    console.error('Error:', error.message);
    return;
  }

  // 날짜별로 집계
  const byDate = {};
  data.forEach(d => {
    const date = d.measured_at.split('T')[0];
    if (!byDate[date]) byDate[date] = [];
    byDate[date].push(d);
  });

  Object.keys(byDate).sort().reverse().forEach(date => {
    const stats = byDate[date];
    const totalPlay = stats.reduce((sum, s) => sum + (s.play_count || 0), 0);
    const postCount = new Set(stats.map(s => s.sponsored_posts?.id)).size;
    
    console.log(`📅 ${date}: ${postCount}개 포스트, 총 조회수 ${totalPlay.toLocaleString()}`);
    
    // 조회수 상위 5개
    const top5 = stats
      .filter(s => s.play_count > 0)
      .sort((a, b) => (b.play_count || 0) - (a.play_count || 0))
      .slice(0, 5);
    
    top5.forEach(s => {
      console.log(`  └ [${s.sponsored_posts?.account_name || '?'}] ${s.play_count?.toLocaleString() || 0}`);
    });
    console.log('');
  });
})().catch(console.error);
